File: src/test/java/no/nav/sbl/dialogarena/mininnboks/consumer/DefaultPersonServiceTest.java
package no.nav.sbl.dialogarena.mininnboks.consumer;

import no.nav.common.auth.SubjectHandler;
import no.nav.tjeneste.virksomhet.person.v3.binding.HentGeografiskTilknytningPersonIkkeFunnet;
import no.nav.tjeneste.virksomhet.person.v3.binding.PersonV3;
import no.nav.tjeneste.virksomhet.person.v3.feil.PersonIkkeFunnet;
import no.nav.tjeneste.virksomhet.person.v3.informasjon.Kommune;
import no.nav.tjeneste.virksomhet.person.v3.meldinger.HentGeografiskTilknytningRequest;
import no.nav.tjeneste.virksomhet.person.v3.meldinger.HentGeografiskTilknytningResponse;
import org.junit.Before;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.mockito.Mock;
import org.mockito.runners.MockitoJUnitRunner;

import static no.nav.sbl.dialogarena.mininnboks.TestUtils.MOCK_SUBJECT;
import static org.hamcrest.core.Is.is;
import static org.junit.Assert.assertThat;
import static org.mockito.Matchers.any;
import static org.mockito.Mockito.when;

@RunWith(MockitoJUnitRunner.class)
public class DefaultPersonServiceTest {
    @Mock
    private PersonV3 personV3;
    private PersonService.Default personService;

    @Before
    public void setUp() {
        personService = new PersonService.Default(personV3);
    }

    @Test
    public void henterEnhet() throws Exception {
        String enhet = "1234";

        when(personV3.hentGeografiskTilknytning(any(HentGeografiskTilknytningRequest.class)))
                .thenReturn(new HentGeografiskTilknytningResponse().withGeografiskTilknytning(new Kommune().withGeografiskTilknytning(enhet)));

        SubjectHandler.withSubject(MOCK_SUBJECT, () -> {
            assertThat(personService.hentGeografiskTilknytning().get(), is(enhet));
        });
    }

    @Test(expected = RuntimeException.class)
    public void kasterRuntimeExceptionOmEnhetIkkeKanhentes() {

        personService.hentGeografiskTilknytning();
    }

}

File: src/test/kotlin/no/nav/sbl/dialogarena/mininnboks/consumer/DefaultPersonServiceTest.kt
package no.nav.sbl.dialogarena.mininnboks.consumer

import io.mockk.coEvery
import io.mockk.mockk
import kotlinx.coroutines.runBlocking
import no.nav.sbl.dialogarena.mininnboks.TestUtils.dummySubject
import no.nav.tjeneste.virksomhet.person.v3.binding.PersonV3
import no.nav.tjeneste.virksomhet.person.v3.informasjon.Kommune
import no.nav.tjeneste.virksomhet.person.v3.meldinger.HentGeografiskTilknytningResponse
import org.hamcrest.MatcherAssert.assertThat
import org.hamcrest.core.Is
import org.spekframework.spek2.Spek

val personV3: PersonV3 = mockk()

object DefaultPersonServiceTest : Spek({

    var personService: PersonService.Default = PersonService.Default(personV3)

    test("henter Enhet") {
        val enhet = "1234"
        coEvery { personV3.hentGeografiskTilknytning(any()) } returns
            HentGeografiskTilknytningResponse().withGeografiskTilknytning(Kommune().withGeografiskTilknytning(enhet))
        runBlocking {
            assertThat(personService.hentGeografiskTilknytning(dummySubject()), Is.`is`(enhet))
        }
    }

    test("`kaster Runtime Exception Om Enhet Ikke Kan hentes") {
        coEvery { personV3.hentGeografiskTilknytning(any()) } throws RuntimeException("RuntimeException")
        try {
            runBlocking {
                personService.hentGeografiskTilknytning(dummySubject())
            }
        } catch (t: Throwable) {
            assertThat(t.message, Is.`is`("RuntimeException"))
        }
    }
})
